### PR TITLE
Add API documentation

### PR DESCRIPTION
**Pull request by peterychang**
Adding docs for a few missing/sparse APIs

### CODE REVIEW

File: python/packages/autogen-core/src/autogen_core/_cancellation_token.py [hunk 1]
```diff
@@ -4,30 +4,36 @@
...
 class CancellationToken:
+    """A token used to cancel pending async calls"""
+
     def __init__(self) -> None:
         self._cancelled: bool = False
         self._lock: threading.Lock = threading.Lock()
         self._callbacks: List[Callable[[], None]] = []
 
     def cancel(self) -> None:
+        """Cancel a pending async call"""
```
Comment (ekzhu) [inline python/packages/autogen-core/src/autogen_core/_cancellation_token.py:15]:
```suggestion
        """Cancel pending async calls linked to this cancellation token."""
```

**Review by ekzhu — APPROVED** [review]

File: python/packages/autogen-core/src/autogen_core/_agent_id.py [hunk 2]
```diff
@@ -44,8 +51,18 @@ def from_str(cls, agent_id: str) -> Self:
 
     @property
     def type(self) -> str:
+        """
+        An identifier that associates an agent with a specific factory function.
+
+        Strings may only be composed of alphanumeric letters (a-z) and (0-9), or underscores (_).
```
Comment (jackgerrits) [inline python/packages/autogen-core/src/autogen_core/_agent_id.py:57]: This is no longer accurate - see: https://github.com/microsoft/autogen/blob/main/docs/design/02%20-%20Topics.md#identifiers

**Review by jackgerrits — COMMENTED** [review]
couple small comments

File: python/packages/autogen-core/src/autogen_core/_agent_id.py [hunk 3]
```diff
@@ -44,8 +51,18 @@ def from_str(cls, agent_id: str) -> Self:
...
+        """
         return self._type
 
     @property
     def key(self) -> str:
+        """
+        Agent instance identifier.
+
+        Strings may only be composed of alphanumeric letters (a-z) and (0-9), or underscores (_).
```
[the remark this answers is not in the document]
Comment (peterychang) [reply python/packages/autogen-core/src/autogen_core/_agent_id.py:66]: This is accurate for AgentId https://microsoft.github.io/autogen/dev/user-guide/core-user-guide/core-concepts/agent-identity-and-lifecycle.html#agent-id

File: python/packages/autogen-core/src/autogen_core/_agent_id.py (hunk 2 above)
Comment (peterychang) [reply python/packages/autogen-core/src/autogen_core/_agent_id.py:57]: This is accurate for AgentId

File: python/packages/autogen-core/src/autogen_core/_agent_id.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (jackgerrits) [reply python/packages/autogen-core/src/autogen_core/_agent_id.py:66]: That doc is also wrong. The truth is in the code https://github.com/microsoft/autogen/blob/9c8877ed664e3ec7bb3896030f4cf0dfbf38558e/python/packages/autogen-core/src/autogen_core/_agent_id.py#L9

**Pull request merged**